### ISSUE TITLE
Have `BatchNorm` handle multiple vmaps when they all have the same `axis_name`.

### ISSUE THREAD

**Issue by patrick-kidger** [patrick-kidger/equinox#49]
At the moment

```python
bn = eqx.nn.BatchNorm("batch", 5)
x = jr.normal(key, (5,))
jax.vmap(jax.vmap(bn, axis_name="batch"), axis_name="batch)(x)
```

will raise weird and wonderful errors.

In practice this should probably still be an error because I don't think this is a concept that makes sense in JAX, but ideally we'd be a bit clearer about what the problem is.

**Comment by patrick-kidger** [comment]
Closing as I think this isn't a well-defined concept in JAX, and it is multiple to handle multiple vmaps like so:
```python
bn = eqx.experimental.BatchNorm(5, ("x", "y"))
jax.vmap(jax.vmap(bn, axis_name="x"), axis_name="y")
```
